be2-scala/src/main/scala/ch/epfl/pop/Server.scala

```diff
@@ -46,16 +46,24 @@ object Server {
       val pubSubMediatorRef: ActorRef = system.actorOf(PubSubMediator.props, "PubSubMediator")
       val dbActorRef: AskableActorRef = system.actorOf(Props(DbActor(pubSubMediatorRef, messageRegistry)), "DbActor")
 
-      def publishSubscribeRoute: RequestContext => Future[RouteResult] = path(config.path) {
-        handleWebSocketMessages(PublishSubscribe.buildGraph(pubSubMediatorRef, dbActorRef, messageRegistry)(system))
+      // Setup routes
+      def publishSubscribeRoute: RequestContext => Future[RouteResult] = {
+        path(config.clientPath) {
+          handleWebSocketMessages(PublishSubscribe.buildGraph(pubSubMediatorRef, dbActorRef, messageRegistry)(system))
+        } ~ path(config.serverPath) {
+          handleWebSocketMessages(PublishSubscribe.buildGraph(pubSubMediatorRef, dbActorRef, messageRegistry)(system))
+        }
```

Comment:
As both routes are the same they can be concatenated using `(path(config.clientPath) | path(config.serverPath)) {handleWebSocketMessages(...)}`

Reply:
Thats only temporary, the next steps are to change `buildGraph` so it behaves slightly differently based on the path used. More precisely: When `/server` is used, the clientActorRef will be sent to the `ConnectionMediator` (pr #1503 )
And the clientActor will also tell `ConnectionMediator` when it disconnects. It took me some time to figure out the `~`, I figured this way already have our template ready.

Reply:
Ok that makes sense !